cmd/haproxy-cfg-generator/cfg_generator.go

```diff
@@ -212,8 +212,9 @@ func get_backends(config *Config) string {
 	}
 
 	if config.tlsMode == BRIDGING {
-		tp_backend = "backend tenant_api_%v\n    server tp_%v %v:%v maxconn %v ssl check ca-file /etc/haproxy/ca.crt\n\n"
-		rp_backend = "backend resource_api\n    server rp %v:%v maxconn %v ssl check ca-file /etc/haproxy/ca.crt\n\n"
+		// perform a TCP check for backend health
+		tp_backend = "backend tenant_api_%v\n    option tcp-check\n    tcp-check connect\n    server tp_%v %v:%v maxconn %v ssl check ca-file /etc/haproxy/ca.crt\n\n"
+		rp_backend = "backend resource_api\n    option tcp-check\n    tcp-check connect\n    server rp %v:%v maxconn %v ssl check ca-file /etc/haproxy/ca.crt\n\n"
 	} else {
```

Comment:
For the configs in the "else" clause, do we need health check or by default it's already tcp-check?

Reply:
per the HAPROXY config doc,  if not set, it will not perform health check and assumed the backend service is always available. we probably will need add if we continue to use the insecure mode

"- when not set, no health checking is performed, and the server is always
    considered available."

Reply:
we will add when needed. from now on, we will be testing secure mode as main stream of testing.